\name{DoEpiMod}
\alias{DoEpiMod}
%- Also NEED an '\alias' for EACH other topic documented here.
\title{
Identifies differential DNA methylation network hotspots.
}
\description{
This function aims to identify subnetworks where many members exhibit
differential DNA methylation in relation to the phenotype of interest.
}
\usage{
DoEpiMod(intEpi.o, nseeds = 100, gamma = 0.5, nMC = 1000,
 sizeR.v = c(1, 100), minsizeOUT = 10, writeOUT = TRUE, nameSTUDY = "X", ew.v = NULL)
}
%- maybe also 'usage' for other objects documented here.
\arguments{
  \item{intEpi.o}{The output of the DoIntEpi450k function}
  \item{nseeds}{An integer specifying the number of seeds and therefore
  modules to search for. By default this number is 100.}

  \item{gamma}{A parameter of the spin-glass algorithm, which determines
  the average module size. Default value generally leads to modules in
  the desired size range (10-100 genes).}

  \item{nMC}{Number of Monte Carlo runs for establishing statistical significance of modularity values under randomisation of the molecular profiles on the network.}

  \item{sizeR.v}{Desired size range for modules.}
  
  \item{minsizeOUT}{
 Minimum size of modules to report as interesting.
}

  \item{writeOUT}{A logical to indicate whether to write out tables in text format.}

  \item{nameSTUDY}{A name for the study, to be used as label in the
    output files.}

  \item{ew.v}{The edge weight vector of the integrated network. This is
    actually generated by the function itself, and can speed up
    inference significantly, if provided as argument to a 2nd instance
    of the function. Default value is NULL.}
  
}
\details{
%%  ~~ If necessary, more details than the description above ~~
}
\value{
A list with following entries:

\item{size}{A vector of inferred module sizes for each of the ntop seeds.}
\item{mod}{A vector of associated modularities.}
\item{pv}{A vector of associated significance P-values with resolution
  limited by nMC runs.}
\item{selmod}{Index positions of significant modules of size at least minsizeOUT}
\item{fem}{A summary matrix of the selected modules.}
\item{topmod}{A list of summary matrices for each of the selected
  modules.}
\item{sgc}{A list of the spin-glass module detection algorithm for each
  seed.}
\item{ew}{The edge-weight vector of the integrated network.}
\item{adj}{ adjacency matrix of the maximally connected integrated
  network (at present only maximally connected subnetwork is used).It is same to intEpi.o$adj, and wil be used for FemModShow function}
}
\references{
A systems-level integrative framework for genome-wide DNA methylation and gene expression data identifies differential gene expression modules under epigenetic control. Jiao Y, Widschwendter M, Teschendorff AE. Bioinformatics. 2014;30(16):2360-2366
}
\author{
"Yinming Jiao"<20907099@zju.edu.cn>, "Andrew E Teschendorff"<andrew@picb.ac.cn>
}

\note{
%%  ~~further notes~~
}

%% ~Make other sections like Warning with \section{Warning }{....} ~

\seealso{
%% ~~objects to See Also as \code{\link{help}}, ~~~
}
\examples{
data(Toydata);
intEpi.o <- list(statM=Toydata$statM,adj=Toydata$adjacency);
EpiMod.o=DoEpiMod(intEpi.o,nseeds=1,gamma=0.5,nMC=1000,sizeR.v=c(1,100),
 minsizeOUT=10,writeOUT=TRUE,nameSTUDY="TEST",ew.v=NULL);

#You can also test on the Realdata which contains DNA methylation of 17
#normal and 118 endometrial cancer samples. Since running on the realdata is time-consuming,  we comment it out.   
#data(Realdata);
# intEpi.o <- list(statM=Realdata$statM,adj=Realdata$adjacency);
#EpiMod.o=DoEpiMod(intEpi.o,nseeds=100,gamma=0.5,nMC=1000,sizeR.v=c(1,100),nameSTUDY="TEST")
}
% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory.
\keyword{ ~kwd1 }
\keyword{ ~kwd2 }% __ONLY ONE__ keyword per line
