\name{DoIntExp}
\alias{DoIntExp}
\title{
Integrates statistics of differential expression for a specified
contrast with a network adjacency matrix.
}
\description{
Integrates statistics of differential expression for a specified
contrast with a network adjacency matrix. Output of this function is
required input for DoExpMod.
}
\usage{
DoIntExp(statR.o,adj.m,c)
}
%- maybe also 'usage' for other objects documented here.
\arguments{
 \item{statR.o}{
The output of GenStatR, or analogous object generated by user.
}
  \item{adj.m}{
The adjacency matrix defining the network of gene relations, e.g. a PPI
network, annotated to NCBI entrez gene IDs.
}
  \item{c}{An integer specifying the desired contrast, i.e. column of
    the contrast matrix. User needs to check which column of statR.o$cont is the one of interest.}
}


\details{
%%  ~~ If necessary, more details than the description above ~~
}
\value{
A list with following entries:
\item{statR}{matrix of gene expression moderated t-statistics and P-values for the genes in the integrated network}
\item{adj}{adjacency matrix of the maximally connected integrated
  network (at present only the maximally connected subnetwork is used
  for subsequent inference)}
\item{avexp}{average expression data matrix mapped to unique Entrez IDs}

}
\references{
A systems-level integrative framework for genome-wide DNA methylation and gene expression data identifies differential gene expression modules under epigenetic control. Jiao YM, Widschwendter M, Teschendorff AE. Bioinformatics. 2014;30(16):2360-2366
}
\author{
"Yinming Jiao"<20907099@zju.edu.cn>, "Andrew E Teschendorff"<andrew@picb.ac.cn>
}
\note{
%%  ~~further notes~~
}

%% ~Make other sections like Warning with \section{Warning }{....} ~

\seealso{
%% ~~objects to See Also as \code{\link{help}}, ~~~
}
\examples{
}
% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory.
\keyword{ ~kwd1 }
\keyword{ ~kwd2 }% __ONLY ONE__ keyword per line
