\name{FemModShow}
\alias{FemModShow}
%- Also NEED an '\alias' for EACH other topic documented here.
\title{
Generates a figure showing the inferred FEM module(s)
}
\description{
This function generates a network representation of an inferred FEM module.
}
\usage{
FemModShow(mod, name, fem.o, mode="Integration")
}
%- maybe also 'usage' for other objects documented here.
\arguments{
  \item{mod}{
the module of the FEM result object for which a figure is to be generated
}
  \item{name}{
the name for the output figure file
}

  \item{fem.o}{
FEM result object generated by DoFEMbi, DoEpiMod, or DoExpMod
}
 

\item{mode}{
There are three modes: "Integration","Epi","Epx". "Integration" means the module is from DoFEMbi, "Epi" means the module is from DoEpiMod, "Exp" means the module is from DoExpMod. 
%%     ~~Describe \code{Realdata} here~~
}

}
\details{
%%  ~~ If necessary, more details than the description above ~~
}
\value{
An igraph object representing the module. At the same time, the module
graph is generated and saved as a "X.pdf" file with "X" the name given
in the argument to the function. 
}
\references{
A systems-level integrative framework for genome-wide DNA methylation and gene expression data identifies differential gene expression modules under epigenetic control. Jiao YM, Widschwendter M, Teschendorff AE. Bioinformatics. 2014;30(16):2360-2366
}
\author{
"Yinming Jiao"<20907099@zju.edu.cn>, "Andrew E Teschendorff"<andrew@picb.ac.cn>
}
\note{
%%  ~~further notes~~
}

%% ~Make other sections like Warning with \section{Warning }{....} ~

\seealso{
%% ~~objects to See Also as \code{\link{help}}, ~~~
}
\examples{
data(Realdata$fembi.o)
data(Realdata)
FemModShow(Realdata$fembi.o$topmod$HAND2,name="HAND2",Realdata$fembi.o)
% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory.
}
\keyword{ ~kwd1 }
\keyword{ ~kwd2 }% __ONLY ONE__ keyword per line
